icu4c/source/i18n/dtfmtsym.cpp

```diff
@@ -1688,14 +1688,11 @@ struct CalendarDataSink : public ResourceSink {
             if (value.getType() == URES_STRING) {
                 // We are on a leaf, store the map elements into the stringMap
                 if (i == 0) {
-                    LocalPointer<Hashtable> stringMapPtr(new Hashtable(FALSE, errorCode), errorCode);
-                    stringMap = stringMapPtr.getAlias();
+                    // mapRefs will keep ownership of 'stringMap':
+                    stringMap = mapRefs.create(FALSE, errorCode);
+                    if (stringMap == NULL) { return; }
```

Comment:
Sorry, one more thing: The old code also set an error if the allocation failed. In this version you need
```
if (stringMap == NULL) {
    errorCode = U_MEMORY_ALLOCATION_ERROR;
    return;
}
```

Reply:
Done.